include/openssl/x509v3.h

```diff
@@ -862,8 +862,50 @@ int X509v3_addr_validate_resource_set(STACK_OF(X509) *chain,
                                       IPAddrBlocks *ext, int allow_inheritance);
 
 #endif                         /* OPENSSL_NO_RFC3779 */
+    
+typedef struct NamingAuthority_st {
+    ASN1_OBJECT* namingAuthorityId;
+    ASN1_IA5STRING* namingAuthorityUrl;
+    ASN1_STRING* namingAuthorityText;          /* i.e. DIRECTORYSTRING */
+} NAMING_AUTHORITY;
+
+DECLARE_ASN1_ITEM(NAMING_AUTHORITY)
+
+typedef struct ProfessionInfo_st {
+    NAMING_AUTHORITY* namingAuthority;
+    STACK_OF(ASN1_STRING)* professionItems;    /* i.e. DIRECTORYSTRING */
+    STACK_OF(ASN1_OBJECT)* professionOIDs;
+    ASN1_PRINTABLESTRING* registrationNumber;
+    ASN1_OCTET_STRING* addProfessionInfo;
+} PROFESSION_INFO;
+
+DECLARE_ASN1_ITEM(PROFESSION_INFO)
+
+typedef struct Admissions_st {
+    GENERAL_NAME* admissionAuthority;
+    NAMING_AUTHORITY* namingAuthority;
+    STACK_OF(PROFESSION_INFO)* professionInfos;
+} ADMISSIONS;
+
+DECLARE_ASN1_ITEM(ADMISSIONS)
+
+typedef struct AdmissionSyntax_st {
+    GENERAL_NAME* admissionAuthority;
+    STACK_OF(ADMISSIONS)* contentsOfAdmissions;
+} ADMISSION_SYNTAX;
+
+DECLARE_ASN1_ITEM(ADMISSION_SYNTAX)
+
+DECLARE_ASN1_FUNCTIONS(NAMING_AUTHORITY)
```

Comment:
May be you could try to use `DECLARE_ASN1_FUNCTIONS_const` instead of `DECLARE_ASN1_FUNCTIONS`